spec/saver_spec.rb

```diff
@@ -46,7 +46,7 @@
       bin = bin_path('pie.elf')
       with_tempfile do |tmp|
         described_class.new(bin, tmp, needed: %w[a.so b.so]).save!
-        expect(PatchELF::Patcher.new(tmp).needed).to eq %w[a.so b.so]
+        expect(PatchELF::Patcher.new(tmp).needed).to match_array %w[a.so b.so]
```

Comment:
@david942j 
i confirmed with @woodruffw  that 2 matching needed are equal regardless of order.
He claims returning in ELF file offset order makes sense to him.

Anyway, i ignored his second statement and
replaced `.eq` with `match_array` as the new implementation makes it return it in different order
but the content is same.

Reply:
It's fine, but I don't understand why this change is needed though (see my comments in lib/patchelf/saver.rb)

Reply:
restored it to original state after the new changes.